_posts/2018-07-10-react-native-wdio.md

```diff
@@ -0,0 +1,358 @@
+---
+layout: post
+title: "React Native Application UI testing using WebDriverIO and Appium"
+date: 2018-07-09 22:58:15 -0800
+cover: /assets/images/cover-wdio.png
+excerpt: We recently adopted WebDriverIO based UI testing for our React Native application. Benefits of using WebDriverIO include allowing us to write UI tests just as we wrote tests for the web. The [synchronous nature of WebDriverIO](http://webdriver.io/guide/getstarted/v4.html#It%E2%80%99s-all-synchronous) helps us write clean UI tests with PageObject models. WebDriverIO configuration allows us to plugin SauceLabs Emulators or Real Devices for cloud-based testing.
+authors:
+  - name: Raja Panidepu
+    url: https://www.linkedin.com/in/rpanidepu/
+    photo: /assets/images/rpanidepu.jpg
+---
+
+## Motivation:
+I work on a Mobile App team which employs the React Native Framework. Just as we wrote
+JavaScript code to develop React Native components for the Android and iOS platforms, we
+wanted to write our UI tests in a similar fashion where we are able to re-use the tests
+across iOS and Android platforms. In order to achieve that we explored few options and the
+final answer we found was WebDriverIO + Appium.
```

Comment:
Can you add some discussion about what else you considered and why you didn't choose it?

Reply:
Trying to keep it to the point, instead of introducing more content.